File: ios/RCTBaiduMap/BaiduBridgeHelper.m
//
//  BaiduBridgeHelper.m
//  RCTBaiduMap
//
//  Created by Sasha Prokhorenko on 24.10.17.
//  Copyright © 2017 Sasha Prokhorenko All rights reserved.
//

#import "BaiduBridgeHelper.h"
#import <React/RCTLog.h>
#import "Constants.h"

@implementation BaiduBridgeHelper

@synthesize bridge = _bridge;

RCT_EXPORT_MODULE(BaiduBridgeHelper)

RCT_EXPORT_METHOD(setPinImageWith:(NSString *)imageName) {
  UIImage *image = [UIImage imageWithContentsOfFile: imageName];
  if (image != nil) {
    NSDictionary *userInfo = [NSDictionary dictionaryWithObject:image forKey:pinImageKey];
    [[NSNotificationCenter defaultCenter] postNotificationName:sendCustomPinImage object:self userInfo:userInfo];
  }
}

RCT_EXPORT_METHOD(setCustomMapStyleWith:(NSString *)JSONFile) {
  NSDictionary *userInfo = [NSDictionary dictionaryWithObject:JSONFile forKey:jsonFileKey];
  [[NSNotificationCenter defaultCenter] postNotificationName: customMapStyleWithJSON object:self userInfo: userInfo];
}

RCT_EXPORT_METHOD(customMapStyle:(BOOL)enable) {
  NSDictionary *userInfo = [NSDictionary dictionaryWithObject: [NSNumber numberWithBool:enable]  forKey: customMapKey];
  [[NSNotificationCenter defaultCenter] postNotificationName: useCustomMapStyle object:self userInfo: userInfo];
}

@end

File: android/src/main/java/com/baidumap/BaiduBridgeHelper.java
package com.baidumap;

import com.facebook.react.bridge.Arguments;
import com.facebook.react.bridge.ReactApplicationContext;
import com.facebook.react.bridge.ReactMethod;
import com.facebook.react.bridge.WritableMap;

/**
 * Created by Sasha Prokhorenko on 27.10.17.
 */

public class BaiduBridgeHelper extends BaseModule {

    public BaiduBridgeHelper(ReactApplicationContext reactContext) {
        super(reactContext);
        context = reactContext;
    }

    public String getName() {
        return "BaiduBridgeHelper";
    }

    @ReactMethod
    public void setPinImageWith(String imageName) {

    }

    @ReactMethod
    public void setCustomMapStyleWith(String JSONFile) {

    }


    @ReactMethod
    public void customMapStyle(Boolean enable) {

    }

}
